### PR TITLE
[CodeWhisperer] Add cwspr support for JSX file

### PR DESCRIPTION
**Pull request by Will-ShaoHua**
https://user-images.githubusercontent.com/96078566/189084639-3096effd-8575-44bb-bce3-735413ba66f8.mov

jsx-serviceInvocation+userdecision

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist

- [ ] I have read the **README** document
- [ ] I have read the **CONTRIBUTING** document
- [ ] Local run of `gradlew check` succeeds
- [ ] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed
- [ ] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if change is customer facing in the IDE.
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/language/CodeWhispererLanguageManager.kt [hunk 1]
```diff
@@ -4,15 +4,43 @@
...
+    private val supportedLanguage = setOf(
+        CodewhispererLanguage.Java.toString(),
+        CodewhispererLanguage.Python.toString(),
+        CodewhispererLanguage.Javascript.toString()
+    )
+
+    fun isLanguageSupported(language: ProgrammingLanguage): Boolean {
+        val mappedLanguage = getParentLanguage(language)
+        return supportedLanguage.contains(mappedLanguage.languageName)
+    }
```
Comment (Will-ShaoHua) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/language/CodeWhispererLanguageManager.kt:20]: HelloCpp
HelloJava
390" alt="HelloJSX" src="https://user-images.githubusercontent.com/96078566/189083500-3e26eafd-5515-460f-a9b1-290abcc9b812.png">
-
HelloJS
8035-054c933549f7.png">
HelloPython
HelloTypeScript

Comment (Will-ShaoHua) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/language/CodeWhispererLanguageManager.kt:20]: HelloCpp
HelloJava
HelloJS
HelloJSX
HelloKotlin
HelloPython
HelloTypeScript

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/language/CodeWhispererLanguageManager.kt [hunk 2]
```diff
@@ -4,15 +4,43 @@
...
         fun getInstance(): CodeWhispererLanguageManager = service()
     }
 }
+
+fun ProgrammingLanguage.toCodeWhispererLanguage() = when {
+    languageName.contains("python") -> CodewhispererLanguage.Python
+    languageName.contains("javascript") -> CodewhispererLanguage.Javascript
+    languageName.contains("java") -> CodewhispererLanguage.Java
+    languageName.contains("jsx") -> CodewhispererLanguage.Jsx
+    languageName.contains("""plain.*text""".toRegex()) -> CodewhispererLanguage.Plaintext
```
Comment (andrewyuq) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/language/CodeWhispererLanguageManager.kt:42]: Why are we having this regex? What are the patterns we are seeing for plain text?

**Review by andrewyuq — APPROVED** [review]

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/language/CodeWhispererLanguageManager.kt (hunk 2 above)
Comment (Will-ShaoHua) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/language/CodeWhispererLanguageManager.kt]: actually just let me revert it back to a string literal `plain_text` would be fine as well

**Review by manodnyab — APPROVED** [review]

**Pull request merged**